include/sliding_window/search/query_result.hpp

```diff
@@ -18,13 +18,13 @@ class query_result
     query_result & operator=(query_result &&) = default;
     ~query_result() = default;
 
-    query_result(std::string const query_id, std::set<size_t> query_hits)
+    query_result(std::string const query_id, std::set<size_t> const & query_hits)
     {
         id = query_id;
-        bin_hits = query_hits; 
+        bin_hits = std::move(query_hits); 
```

Comment:
This is weird. As you can't move a const object.

You either give the object by const ref i.e. `std::set<size_t> const & query_hits` and then copy it (not moving `bin_hits = query_hits;`), or you don't qualify the type i.e, `std::set<size_t> query_hits` and move `bin_hits = std::move(query_hits);` it in.

I.e.:
```c++
query_result(std::string const query_id, std::set<size_t> query_hits)
//...
bin_hits = std::move(query_hits); 
```

Reply:
Either way is fine :)

Reply:
Thanks for the explanation :)